spec/px/service/client/base_spec.rb

```diff
@@ -45,5 +45,83 @@
         expect(resp.request.url).to include("one=a&two=b")
       end
     end
+
+    context "when requests fail" do
+      let(:multi) { Px::Service::Client::Multiplexer.new }
+      let(:response) { Typhoeus::Response.new(code: 500, body: "Internal Server Error") }
+
+      before { Typhoeus.stub(url).and_return(response) }
+
+      context "after one request" do
+        let(:request) do
+          multi.context do
+            resp = multi.do(subject.send(:make_request, :get, url))
+            expect{resp.body}.to raise_error(Px::Service::ServiceError)
+          end.run
+        end
+
+        it "does not trip circuit breaker" do
+          request
+          expect(subject.circuit_state.closed?).to be_truthy
+        end
+
+        it "increases the failure count by 1" do
+          expect{
+            request
+          }.to change{subject.circuit_state.failure_count}.by(1)
+        end
+      end
+
+      context "after multiple requests" do
+        let(:handler) { subject.class.circuit_handler }
+        let(:limit) { handler.failure_threshold }
+        let(:timeout) { handler.failure_timeout }
+
+        before { handler.logger = Logger.new(STDOUT) }
+
+        before :each do
+          # trips the circuit breaker
+          (limit + 1).times do
+            multi.context do
+              req = subject.send(:make_request, :get, url)
+              resp = multi.do(req)
+              expect{resp.body}.to raise_error
+            end.run
+          end
+        end
+
+        it "trips the circuit breaker after the limit" do
+          expect(subject.circuit_state.open?).to be_truthy
+
+          subject.circuit_state.reset
+        end
+
+        it "raises CircuitBrokenException after limit" do
+          expect{
+            multi.context do
+              req = subject.send(:make_request, :get, url)
+              multi.do(req)
+            end.run
+          }.to raise_error(CircuitBreaker::CircuitBrokenException)
```

Comment:
The actual behaviour should be to raise a `ServiceError` with the circuit-broken message.  That's the contract that this gem provides to its users.